python/lsst/pipe/supertask/superTask.py

```diff
@@ -134,6 +165,70 @@ def getOutputDatasetTypes(cls, config):
                 dsTypes[key] = cls.makeDatasetType(value)
         return dsTypes
 
+    @classmethod
+    def getInitInputDatasetTypes(cls, config):
+        """Return dataset types that can be used to retrieve the ``initInputs``
+        constructor argument.
+
+        Datasets used in initialization may not be associated with any
+        DataUnits (i.e. their data IDs must be empty dictionaries).
+
+        Default implementation finds all fields of type
+        `InputInputDatasetConfig` in configuration (non-recursively) and uses
+        them for constructing `DatasetType` instances. The keys of these
+        fields are used as keys in returned dictionary. Subclasses can
+        override this behavior.
+
+        Parameters
+        ----------
+        config : `Config`
+            Configuration for this task. Typically datasets are defined in
+            a task configuration.
+
+        Returns
+        -------
+        Dictionary where key is the name (arbitrary) of the input dataset and
+        value is the `butler.core.datasets.DatasetType` instance. Default
+        implementation uses configuration field name as dictionary key.
+        """
+        dsTypes = {}
+        for key, value in config.items():
+            if isinstance(value, InitInputDatasetConfig):
+                dsTypes[key] = cls.makeDatasetType(value)
+        return dsTypes
+
+    @classmethod
+    def getInitOutputDatasetTypes(cls, config):
+        """Return dataset types that can be used to write the objects
+        returned by `getOutputDatasets`.
+
+        Datasets used in initialization may not be associated with any
+        DataUnits (i.e. their data IDs must be empty dictionaries).
+
+        Default implementation finds all fields of type
+        `InitOutputDatasetConfig` in configuration (non-recursively) and uses
+        them for constructing `DatasetType` instances. The keys of these
+        fields are used as keys in returned dictionary. Subclasses can
+        override this behavior.
+
+        Parameters
+        ----------
+        config : `Config`
+            Configuration for this task. Typically datasets are defined in
+            a task configuration.
+
+        Returns
+        -------
+        Dictionary where key is the name (arbitrary) of the output dataset and
+        value is the `butler.core.datasets.DatasetType` instance. Default
+        implementation uses configuration field name as dictionary key.
+        """
+        dsTypes = {}
+        for key, value in config.items():
+            if isinstance(value, InitOutputDatasetConfig):
+                dsTypes[key] = cls.makeDatasetType(value)
+        return dsTypes
```

Comment:
We now have 4 methods with ~identical implementation, I'll probably move it to a new method later.

Reply:
It sounds like you've already got a way to fix this in mind and would prefer to do it later.  Please let me know if you'd like me to; I certainly agree that it's a good idea.

Reply:
You can do it too if you prefer to fix it now, no objections from my side. I'll be working on this code anyways and there is nothing broken now so if you want to move forward just leave it to me to work on it later.

Reply:
Ok, I'm happy to leave it to you, then.